src/core/transaction.c

```diff
@@ -1025,7 +1025,7 @@ int transaction_add_job_and_dependencies(
                                 r = transaction_add_job_and_dependencies(tr, JOB_VERIFY_ACTIVE, dep, ret, true, false, false, ignore_order, e);
                                 if (r < 0) {
                                         if (r != -EBADR) /* job type not applicable */
-                                                return r;
+                                                goto fail;
```

Comment:
hmm, for the UNIT_ATOM_PULL_IN_STOP case below I think we should add the same, no?

can you provide a rationale why not?